src/main/java/jenkins/advancedqueue/jobinclusion/strategy/FolderBasedJobInclusionStrategy.java

```diff
@@ -62,19 +66,70 @@ public ListBoxModel getListFolderItems() {
 
 	};
 
+	static public class JobPattern {
```

Comment:
If you want to create a new class, make it describable and use `optionalProperty`. OTOH I am not sure it is required in this case